File: Backend/dgulaf/src/main/java/com/dgulaf/model/LostItem.java
package com.dgulaf.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.Date;

@Data
@Entity
@Table(name = "LostItems")
public class LostItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer lostItemId;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false, length = 100)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private Date lostDate;

    @ManyToOne
    @JoinColumn(name = "location_id", nullable = false)
    private ItemLocation location;

    @Column(nullable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime createdAt;
}

File: Frontend/dgu_laf/lib/model/lost_item.dart
class LostItem {
  final int lostItemId;
  final int userId;
  final String title;
  final String description;
  final String lostDate;
  final int locationId;
  final String createdAt;

  LostItem({
    required this.lostItemId,
    required this.userId,
    required this.title,
    required this.description,
    required this.lostDate,
    required this.locationId,
    required this.createdAt,
  });

  factory LostItem.fromJson(Map<String, dynamic> json) {
    return LostItem(
      lostItemId: json['lost_item_id'],
      userId: json['user_id'],
      title: json['title'],
      description: json['description'],
      lostDate: json['lost_date'],
      locationId: json['location_id'],
      createdAt: json['created_at'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'lost_item_id': lostItemId,
      'user_id': userId,
      'title': title,
      'description': description,
      'lost_date': lostDate,
      'location_id': locationId,
      'created_at': createdAt,
    };
  }
}
